frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts

```diff
@@ -82,16 +84,44 @@ export const getBarLabelLayout =
     };
   };
 
+export const getBarInsideLabelLayout =
+  (
+    dataset: ChartDataset,
+    settings: ComputedVisualizationSettings,
+    seriesDataKey: DataKey,
+  ): BarSeriesOption["labelLayout"] =>
+  params => {
+    const { dataIndex, rect, labelRect } = params;
+    if (dataIndex == null) {
+      return {};
+    }
+
+    const canFitWithoutRotation =
+      rect.width > labelRect.width && rect.height > labelRect.height;
+
+    if (!canFitWithoutRotation) {
+      return {
+        fontSize: 0,
+      };
+    }
+
+    const labelValue = dataset[dataIndex][seriesDataKey];
```

Comment:
Should use `transformedDataset[ORIGINAL_INDEX_DATA_KEY]` for consistency with `buildEChartsStackLabelOptions`, or in `buildEChartsStackLabelOptions` just use `params.dataIndex` for consistency with this function